### PR TITLE
Fix 7709 | Fix showing keyboard when changing placeholder in entry focused event

### LINKED ISSUE #7709: [Bug] On android, not showing keyboard when changing placeholder in entry focused event
### Description
On Android, the keyboard does not appear when I change the PlaceHolder property of a in entry's focusing event. 
### Steps to Reproduce

Entry x:Name="entTest" Placeholder="blabla" Focused="focus_event" 

public void focus_event(object sender, FocusRequestArgs arg)
{
   entTest.Placeholder = null;
}
### Basic Information

- Version with issue: Xamarin.Forms 4.2.0.815419
- Last known good version:
- IDE: VS for Mac 8.3
- Platform Target Frameworks: Android 10.0
- Android:
    Min      :  Andoid 5.0 api level 21
     Target : Android 10.0 api levelt 29

### PR DESCRIPTION
**Pull request by NordAlex**
### Description of Change ###
I handle setting of placeholder for entry. As I see described keyboard hiding persists on native Android development. So to conduct development experience on both platforms I handle it on Android renderer.

### Issues Resolved ### 
- fixes #7709

### API Changes ###
None

### Platforms Affected ### 
- Android

### Behavioral/Visual Changes ###
A keyboard will be shown correctly with Focused handler.

### Before/After Screenshots ### 
Before
before_changes

After
afterChanges

### Testing Procedure ###
1. Add entry;
2. Add a handler for Focused event;
3. Set placeholder in event handler method;
4. Click on entry. You will notice that the text field has focus but the keyboard does not appear on the screen.

Entry x:Name="entTest" Placeholder="blabla" Focused="focus_event"

public void focus_event(object sender, FocusRequestArgs arg)
{
entTest.Placeholder = null;
}

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by NordAlex** [comment]
@samhouts I contribute as a member of hacktoberfest. Could you please add relevant label?

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7709.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+			var entry = new Entry();
+			entry.Placeholder = "Before focus";
+			entry.Focused += Entry_Focused;
+
+			stack.Children.Add(entry);
+
+			Content = stack;
+		}
+
+		private void Entry_Focused(object sender, FocusEventArgs e)
```
Comment (jfversluis) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7709.cs]:
```suggestion
		void Entry_Focused(object sender, FocusEventArgs e)
```

Comment (jfversluis) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7709.cs]: As per the coding guidelines, please do not use the `private` keyword. Thanks!

File: Xamarin.Forms.Platform.Android/Renderers/EntryRenderer.cs [hunk 2]
```diff
@@ -220,7 +220,17 @@ protected override void Dispose(bool disposing)
...
+		{
+			if (EditText.Hint != Element.Placeholder)
+			{
+				EditText.Hint = Element.Placeholder;
+				if (EditText.IsFocused)
+				{
+					EditText.ShowKeyboard();
+				}
+			}
+		}
```
Comment (felipebaltazar) [inline Xamarin.Forms.Platform.Android/Renderers/EntryRenderer.cs]: here you can use the early return pattern

Like this:

```C#
			if (EditText.Hint == Element.Placeholder)
				return;

			EditText.Hint = Element.Placeholder;
			if (EditText.IsFocused)
			{
				EditText.ShowKeyboard();
			}
```

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7709.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+			Content = stack;
+		}
+
+		void Entry_Focused(object sender, FocusEventArgs e)
+		{
+			var entry = sender as Entry;
+			entry.Placeholder = "After focus";
+		}
+
+		protected override void Init()
```
Comment (felipebaltazar) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7709.cs]: so, if you dont use `Init()` you can remove it

Comment (jfversluis) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7709.cs]: No, this is mandatory from the above abstract class

Comment (felipebaltazar) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7709.cs]: hummm, my mistake sorry ..
But maybe can remove the white space below? 😜

File: Xamarin.Forms.Platform.Android/Renderers/EntryRenderer.cs (hunk 2 above)
Comment (NordAlex) [reply Xamarin.Forms.Platform.Android/Renderers/EntryRenderer.cs]: Yes sure. Will change it.

**Comment by NordAlex** [comment]
As I see there is an exception connected to UnitTest library. Should I add some changes to fix it? On my local computer, it isn't reproduced.

**Comment by jfversluis** [comment]
@NordAlex would you be able to make that last change? I think we're close to merging it!

**Comment by NordAlex** [comment]
Yes. I will do it this weekend

**Comment by NordAlex** [comment]
I've updated pr title and get current master branch. Is it ok now?

**Pull request merged**